### PR TITLE
Add GitHub Actions for unit tests and linter

### PR DESCRIPTION
**Pull request by e0ne**
Signed-off-by: Ivan Kolodyazhny <ikolodiazhny@nvidia.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: "Build"
+on: [push, pull_request]
+
+jobs:
+  validate:
```
Comment (adrianchiris) [inline .github/workflows/build.yml:5]: nit: split to 3 jobs ? 

1. build (make build)
2. lint (make lint)
3. test (make test)

Comment (e0ne) [reply .github/workflows/build.yml:5]: I like the idea to make CI faster! I'll do it in a next PR update.

**Comment by e0ne** [comment]
New actions won't run until they will be merged into the repository. A similar PR for network-operator https://github.com/Mellanox/network-operator/pull/218 works because it's created from the branch in the same repository

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      contents: read
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.13'
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: build
+        run: make build
```
Comment (adrianchiris) [inline .github/workflows/build.yml:46]: nit: missing newline

**Comment by e0ne** [comment]
GitHub doesn't run actions proposed in a PR from a fork. Here is a PR which tests these changes: https://github.com/e0ne/k8s-rdma-shared-dev-plugin/pull/1

**Comment by adrianchiris** [comment]
Approved!

**Pull request merged**